### PR TITLE
Simplify DeclarativeSchema API for adding Condition Count Metrics

### PR DESCRIPTION
**Pull request by FelipeAdachi**
## Description

Currently, to add a Resolver for a Condition Count Metric to a DeclarativeSchema, the user needs to define it's own ResolverSpec, like:

```python
date_format_condition = {
    "is_date_format": Condition(Predicate().is_(date_format))
}

date_format_spec = ResolverSpec(
    column_name="last_review",
    metrics=[
        MetricSpec(
            ConditionCountMetric,
            ConditionCountConfig(conditions=date_format_condition),
        ),
    ],
)
schema = DeclarativeSchema(STANDARD_RESOLVER)
schema.add_resolver(date_format_spec)
```

The changes in this PR simplifies this process, enabling the user to call `DeclarativeSchema`'s `add_condition_count_metric`:

```python
  date_format_condition = {
      "is_date_format": Condition(Predicate().is_(date_format))
  }

  schema = DeclarativeSchema(STANDARD_RESOLVER)
  schema.add_condition_count_metric(column_name="last_review", conditions=date_format_condition)
```

`add_condition_count_metric` will basically do the same thing, but internally.

Also:

- Update examples that touch Condition Count Metrics

- [x] I have reviewed the [Guidelines for Contributing](CONTRIBUTING.md) and the [Code of Conduct](CODE_OF_CONDUCT.md).

### CODE REVIEW

**Comment by richard-rogers** [comment]
Consider creating `ConditionCountMetricSpec <: MetricSpec`instead.

**Comment by FelipeAdachi** [comment]
> Consider creating `ConditionCountMetricSpec <: MetricSpec`instead.

@richard-rogers can you confirm if the latest change address what you had in mind?

File: python/whylogs/core/resolvers.py [hunk 1]
```diff
@@ -99,6 +103,20 @@ def __post_init__(self):
...
+    Specify a ConditionCountMetric to instantiate.
+    """
+
+    metric: ConditionCountMetric
+    config: Optional[ConditionCountConfig] = None  # omit to use default MetricConfig
+
+    def __post_init__(self):
+        if not issubclass(self.metric, ConditionCountMetric):
+            raise ValueError("ConditionCountMetricSpec: must supply a ConditionCountMetric to MetricSpec")
+
```
Comment (richard-rogers) [inline python/whylogs/core/resolvers.py]: I was thinking something like (in condition_count_metric.py):
```
@dataclass(init=False)  # I'm not sure that's the exact parameter name
class ConditionCountMetricSpec(MetricSpec):
  __init__(self, conditions: Dict[str, Condition], config: Optional[ConditionCountConfig]=None):
    self.metric = ConditionCountMetric
    self.config = config if config else ConditionCountConfig()
    self.config.conditions.update(conditions)
```
Then you could do
```
date_format_condition = {
    "is_date_format": Condition(Predicate().is_(date_format))
}

date_format_spec = ResolverSpec(
    column_name="last_review",
    metrics=[ConditionCoundMetricSpec(date_format_conditon)],
)
schema = DeclarativeSchema(STANDARD_RESOLVER)
schema.add_resolver(date_format_spec)
```
We could also turn `DatasetSchema::add_resovler()` into more of a factory:
```
def add_resolver_spec(self, column_name: Optional[str]=None, column_type: Optional[Any], metrics: List[MetricSpec]):
  spec = MetricSpec(column_name, column_type, metrics)
  self.add_resolver(spec)
```

So we'd end up with
```
date_format_condition = {
    "is_date_format": Condition(Predicate().is_(date_format))
}

schema = DeclarativeSchema(STANDARD_RESOLVER)
schema.add_resolver_spec("last_review", None, [ConditionCoundMetricSpec(date_format_conditon)])
```
I think this is more a more general simplification.

File: python/whylogs/core/schema.py [hunk 2]
```diff
@@ -211,6 +212,12 @@ class DeclarativeSchema(DatasetSchema):
     def add_resolver(self, resolver_spec: ResolverSpec):
         self.resolvers.add_resolver(resolver_spec)
 
+    def add_resolver_spec(
+        self, column_name: Optional[str] = None, column_type: Optional[Any] = None, metrics: List[MetricSpec] = []
```
Comment (richard-rogers) [inline python/whylogs/core/schema.py]:
```suggestion
        self, column_name: Optional[str] = None, column_type: Optional[Any] = None, metrics: Optional[List[MetricSpec]] = None
```

File: python/whylogs/core/schema.py [hunk 3]
```diff
@@ -211,6 +212,12 @@ class DeclarativeSchema(DatasetSchema):
...
+    ):
+        spec = ResolverSpec(column_name=column_name, column_type=column_type, metrics=metrics)
```
Comment (richard-rogers) [inline python/whylogs/core/schema.py]:
```suggestion
        spec = ResolverSpec(column_name=column_name, column_type=column_type, metrics=metrics or [])
```

**Comment by richard-rogers** [comment]
Looks good. One minor fix to the empty list default argument.

**Review by richard-rogers — APPROVED** [review]

**Pull request merged**